src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -51,16 +55,159 @@ graph_status_to_string(int status) {
   }
 }
 
+void
+populate_aie_core(boost::property_tree::ptree _pt,boost::property_tree::ptree& tile, int row, int col)
+{
+  boost::property_tree::ptree pt;
+  std::string aie_data;
+  std::vector<std::string> graph_status;
+  boost::property_tree::ptree tile_array;
+  boost::property_tree::ptree empty_pt;
+  try {
+    pt =  _pt.get_child("aie_core."+std::to_string(col)+"_"+std::to_string(row));
+    tile.put("column", pt.get<uint32_t>("col"));
+    tile.put("row", pt.get<uint32_t>("row"));
+    int count = 0;
+    std::string status;
+    for (auto& node: pt.get_child("core.Status", empty_pt)) {
+      status +=(status.empty()?"":", ")+node.second.data();
+    }
+    if(!status.empty())
+      tile.put("core.status", status);
+
+    for (auto& node: pt.get_child("core.PC", empty_pt))
+      tile.put("core.program_counter", node.second.data());
+    for (auto& node: pt.get_child("core.LR", empty_pt))
+      tile.put("core.link_register", node.second.data());
+    for (auto& node: pt.get_child("core.SP", empty_pt))
+      tile.put("core.stack_pointer", node.second.data());
+    for (auto& node: pt.get_child("dma.Channel_status.MM2S", empty_pt)) {
+      tile.put("dma.channel_status.mm2s.channel_"+std::to_string(count++), node.second.data());
+    }
+
+    count = 0;
+    for (auto& node: pt.get_child("dma.Channel_status.S2MM", empty_pt)) {
+      tile.put("dma.channel_status.s2mm.channel_"+std::to_string(count++), node.second.data());
+    }
+
+    count = 0;
+    std::string mode;
+    for (auto& node: pt.get_child("dma.Mode.MM2S", empty_pt)) {
+      mode = mode+(mode.empty()?"":", ")+node.second.data();
+    }
+    if(!mode.empty())
+      tile.put("dma.mode.mm2s", mode);
+    mode.clear();
+    for (auto& node: pt.get_child("dma.Mode.S2MM", empty_pt)) {
+      mode = mode+(mode.empty()?"":", ")+node.second.data();
+    }
+    if(!mode.empty())
+      tile.put("dma.mode.s2mm", mode);
+    count = 0;
+    for (auto& node: pt.get_child("dma.Queue_size.MM2S", empty_pt)) {

```

Comment:
The same pattern of for retrieving the data is being repeated over and over again.  Please refactor the code to use a helper method or possibly a lamda function.

Reply:
fixed